R/univariate.R

```diff
@@ -28,11 +28,15 @@ setGeneric("median")
         filter = toJSON(zcl(activeFilter(x)), for_query_string = TRUE)
     )
     cube <- CrunchCube(crGET(cubeURL(x), query = query))
-    if (!na.rm && cube$result$measures[[measure_name]][["n_missing"]] > 0) {
-        return(NA_real_)
-    } else {
-        return(as.array(cube))
+    out <- as.array(cube)
+    # --- Convert from NaN to NA because the zz9 doesn't distinguish
+    # --- Try to do it minimally so that if it's really NaN we preserve it
+    # --- but in reality we don't have enough information to know for sure
+    if (!na.rm && any(vapply(out, is.nan, logical(1)))) {
```

Comment:
1. Why do  we need vapply here? Instead of just:

```
if (!na.rm && any(is.nan(out))) {
    out[is.nan(out)] <- NA
}
```

... simplifying further to:

```
if (!na.rm) {
    out[is.nan(out)] <- NA
}
```

2. What do you mean by doing it 'minimally'?

Reply:
Ha, I think that's python making me forget about vectorized functions 😢 

Meaning if you don't  do `na.rm` we're not setting to `NA`, so that some `NaN`s are preserved. I updated the comment.